README.rst

```diff
@@ -8,3 +8,121 @@ A GPL-licensed Python library that facilitates functional testing of quipucords.
+projects. All portions of the test suite expect the config file format.  Any
+changes to the test suite that require changes to the config file format should
+be made with this in mind.
+
+There is an example annotated config file in ``camayoc/example_config.yaml``
+
+Configuring py.test
+^^^^^^^^^^^^^^^^^^
+
+If you are testing machines that use ``https`` but do not have good
+certificates, it can get difficult to read the test results due to warnings
+about insecure connections. If this is a problem for you, an example
+``pytest.ini`` file is provided in the root of the ``camayoc`` repository that
+disables these warnings. Additional configuration is possible through the
+``pytest.ini`` file, these options are documented at
+https://docs.pytest.org/en/latest/reference.html#ini-options-ref
+
+For example, if you always want to run ``py.tests`` in verbose mode, this can
+be done in the ``pytest.ini`` file.
+
+Running the test suite
+^^^^^^^^^^^^
+
+While ``camayoc`` contains test for ``rho`` and ``quipucords``, and these share
+a config file format, testing both ``rho`` and ``quipucords`` simultaneously is
+not a supported use case.
+
+Camayoc uses ``py.test`` as its test driver. The pytest project had excellent
+documentation available at https://docs.pytest.org/en/latest/contents.html
+
+Running tests for ``rho``
+^^^^^
+
+To run all the tests for ``rho``, first have ``rho`` installed and available in
+your ``$PATH``. This can be done by either ``rpm`` install to the system or by
+``pip`` installing ``rho`` from source into your ``camayoc`` virtual
+environment. Then invoke the main test suite from the root directory of
+``camayco`` with::
+
+    py.test camayoc/tests/rho/
+
+There are some additional tests for rho located in ``camayoc/tests/remote/rho``
+that scan all machines listed in the inventory with all available network
+credentials. To run these, you similarly invoke::
+
+    py.test camayoc/tests/remote/rho/
+
+Running tests for ``quipucords`` and ``qpc``
+^^^^^^
+
+To run all the ``quipucords`` tests, first start the ``quipucords`` server and
+set up your config file to point to the server. You should also install ``qpc``
+in such a way that it is available in your ``$PATH``. Then run all API, CLI,
+and UI tests with::
+
+    py.test camayoc/tests/qcs/
+
+To only test the API, CLI, or UI, tell pytest this by extending the path to
+only point to one of these subdirectories::
+    
+    # for API tests only
+    py.test camayoc/tests/qcs/api
+    # for CLI tests only
+    py.test camayoc/tests/qcs/cli
+    # for UI tests only
+    py.test camayoc/tests/qcs/ui
+
+The tests can be further selected by specifying any path or file. Additionally
+you can select tests based on string matching. For example, to run all tests
+with ``create`` in the name and skip any others, run::
+
+    py.test camayoc/tests/qcs/api -k create
+
+Or to select a specific test, type out the whole name of the test.
+
+Contributing
```

Comment:
Often this section is moved into a **CONTRIBUTING.rst**

For example:
https://github.com/quipucords/quipucords/blob/master/CONTRIBUTING.rst

Although I haven't read the above in a while. 😄